workitem_blackbox_test.go

```diff
@@ -131,3 +136,97 @@ func TestListByFields(t *testing.T) {
+	return func(start float64, limit float64, first string, last string, prev string, next string) {
+		count := computeCount(totalCount, int(start), int(limit))
+		repo.ListReturns(makeWorkItems(count), uint64(totalCount), nil)
+
+		_, response := test.ListWorkitem2OK(t, context.Background(), nil, controller, nil, &limit, &start)
+		assertLink(t, "first", first, response.Links.First)
+		assertLink(t, "last", last, response.Links.Last)
+		assertLink(t, "prev", prev, response.Links.Prev)
+		assertLink(t, "next", next, response.Links.Next)
+		assert.Equal(t, float64(totalCount), response.Meta.TotalCount)
+	}
+}
+
+func assertLink(t *testing.T, l string, expected string, actual *string) {
+	if expected == "" {
+		if actual != nil {
+			assert.Fail(t, fmt.Sprintf("link %s should be nil but is %s", l, *actual))
+		}
+	} else {
+		if actual == nil {
+			assert.Fail(t, "link %s should be %s, but is nil", l, expected)
+		} else {
+			assert.True(t, strings.HasSuffix(*actual, expected), "link %s should be %s, but is %s", l, expected, *actual)
+		}
+	}
+}
+
+func computeCount(totalCount int64, start int, limit int) int {
+	if start < 0 || int64(start) >= totalCount {
+		return 0
+	}
+	if int64(start+limit) > totalCount {
+		return int(totalCount - int64(start))
+	}
+	return limit
+}
+
+func makeWorkItems(count int) []*app.WorkItem {
+	res := make([]*app.WorkItem, count)
+	for index := range res {
+		res[index] = &app.WorkItem{
+			ID:     fmt.Sprintf("id%d", index),
+			Type:   "foobar",
+			Fields: map[string]interface{}{},
+		}
+	}
+	return res
+}
+
+func TestPagingLinks(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
+	ts := testsupport.NoTransactionSupport{}
+	repo := &testsupport.WorkItemRepository{}
+	svc := goa.New("TestPaginLinks-Service")
+	assert.NotNil(t, svc)
+	controller := NewWorkitem2Controller(svc, repo, ts)
+
+	pagingTest := createPagingTest(t, controller, repo, 13)
+	pagingTest(2, 5, "page[offset]=0,page[limit]=2", "page[offset]=12,page[limit]=5", "page[offset]=0,page[limit]=2", "page[offset]=7,page[limit]=5")
+	pagingTest(10, 3, "page[offset]=0,page[limit]=1", "page[offset]=10,page[limit]=3", "page[offset]=7,page[limit]=3", "")
+	pagingTest(0, 4, "page[offset]=0,page[limit]=4", "page[offset]=12,page[limit]=4", "", "page[offset]=4,page[limit]=4")
+	pagingTest(4, 8, "page[offset]=0,page[limit]=4", "page[offset]=12,page[limit]=8", "page[offset]=0,page[limit]=4", "page[offset]=12,page[limit]=8")
+
+	pagingTest(16, 14, "page[offset]=0,page[limit]=2", "page[offset]=2,page[limit]=14", "page[offset]=2,page[limit]=14", "")
+	pagingTest(16, 18, "page[offset]=0,page[limit]=16", "page[offset]=0,page[limit]=16", "page[offset]=0,page[limit]=16", "")
+
+	pagingTest(3, 50, "page[offset]=0,page[limit]=3", "page[offset]=3,page[limit]=50", "page[offset]=0,page[limit]=3", "")
+	pagingTest(0, 50, "page[offset]=0,page[limit]=50", "page[offset]=0,page[limit]=50", "", "")
```

Comment:
It would generally be more readable if you moved the arguments into a list of structs as oppose to 'n' arguments.

e.g. 

```go
	type scenario struct {
		Offset int
		Limit  int
		First  string
		Last   string
		Prev   string
		Next   string
	}

	scenarios := []scenario{
		scenario{
			Offset: 2,
			Limit:  5,
			First:  "page[offset]=0,page[limit]=2",
			Last:   "page[offset]=12,page[limit]=5",
			Prev:   "page[offset]=0,page[limit]=2",
			Next:   "page[offset]=7,page[limit]=5"},
		scenario{
			Offset: 2,
			Limit:  5,
			First:  "page[offset]=0,page[limit]=2",
			Last:   "page[offset]=12,page[limit]=5",
			Prev:   "page[offset]=0,page[limit]=2",
			Next:   "page[offset]=7,page[limit]=5"},
	}

	for index, test := range scenarios {
		err := pageTest(test.Offset, test.Limit, test.First, test.Last, test.Prev, test.Next)
		if err != nil {
			t.Errorf("Scenario %v failed: %v", index, err)
		}
	}
```